hazelcast/src/test/java/com/hazelcast/cluster/ClusterClockTest.java

```diff
@@ -0,0 +1,99 @@
+package com.hazelcast.cluster;
+
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.Node;
+import com.hazelcast.instance.TestUtil;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class ClusterClockTest extends HazelcastTestSupport {
+
+    @Before
+    @After
+    public void cleanup() {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void test_start_time_single_node_cluster() throws Exception {
+        Config config = new Config();
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
+        Node node1 = TestUtil.getNode(h1);
+        System.out.println(node1.isMaster());
+        assertNotEquals(Long.MIN_VALUE , node1.getClusterService().getClusterClock().getClusterStartTime());System.out.println();
+    }
+
+    @Test
+    public void all_nodes_should_have_the_same_cluster_start_time() throws Exception {
+        Config config = new Config();
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h2 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h3 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h4 = Hazelcast.newHazelcastInstance(config);
+
```

Comment:
Can't we use `TestHazelcastInstanceFactory.createHazelcastInstanceFactory` here? then cleanup is done automatically...